test/unit/state.spec.js

```
@@ -0,0 +1,191 @@
+  it('stores prototype-collision names as own values', function() {
+    const attributes = { constructor: 'constructor', toString: 'toString' };
+    Object.defineProperty(attributes, '__proto__', {
+      enumerable: true,
+      value: 'prototype'
+    });
+    const state = new State(attributes);
+    const snapshot = state.toJSON();
+
+    expect(snapshot).to.deep.equal(attributes);
+    expect(Object.getPrototypeOf(snapshot)).to.equal(Object.prototype);
+    expect(snapshot.__proto__).to.equal('prototype');
```

ESLint

[error] The '__proto__' property is deprecated.

(no-proto)